src/net/sourceforge/kolmafia/textui/parsetree/AggregateLiteral.java

```diff
@@ -33,13 +33,44 @@
 
 package net.sourceforge.kolmafia.textui.parsetree;
 
+import net.sourceforge.kolmafia.textui.AshRuntime;
+
 public abstract class AggregateLiteral
 	extends AggregateValue
 {
 	protected AggregateValue aggr = null;
 
-        public AggregateLiteral( final AggregateType type )
+	public AggregateLiteral( final AggregateType type )
 	{
 		super( type );
 	}
+
+	@Override
```

Comment:
These stub methods don't seem useful here -- they should presumably remain abstract.

Reply:
They're not stubs. Neither MapLiteral nor ArrayLiteral (both subclasses of AggregateLiteral) implement any of these methods, so they need a "default".

It's either that, or all of this goes in both MapLiteral and ArrayLiteral instead.

Reply:
They're not stubs. Neither MapLiteral nor ArrayLiteral (both subclasses of AggregateLiteral) implement any of these methods, so they need a "default".

It's either that, or all of this goes in both MapLiteral and ArrayLiteral instead.

Reply:
I get the aset / remove being implemented here, as an AggregateLiteral should be immutable. But MapLiteral and ArrayLiteral don't need to know how to access a value, or iterate over keys, or check for presence?

I'm probably misunderstanding what these methods are actually meant to signify.

Reply:
Quickly looking at its two subclasses, it seems like they override a single method: Value.execute().

This method turns them into a *different* AggregateValue (either an ArrayValue or a MapValue), which are *then* acted on.

Reply:
(single, besides Value.count()...)